pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix

```
@@ -107,7 +110,10 @@ let
       ++ lib.optional stdenv.hostPlatform.isLinux autoPatchelfHook;
 
     propagatedBuildInputs = [ setJavaClassPath zlib ]
-      ++ lib.optional stdenv.hostPlatform.isDarwin darwin.apple_sdk_11_0.frameworks.Foundation;
+      ++ lib.optionals (stdenv.hostPlatform.isDarwin && lib.versionOlder apple-sdk.version "11") [
```

Stolen from here: https://github.com/NixOS/nixpkgs/pull/351004#discussion_r1815645876

---

so it turns out that the ` lib.versionOlder apple-sdk.version "11"`  logic is unnecessary. The build will use the highest sdk. so if some other package propagates `apple-sdk_12` then `apple-sdk_12` will be used even if we add `apple-sdk_11` here. or if `apple-sdk` ends up being version 12, we'll use version 12 even if we specify `apple-sdk_11` here.

```suggestion
      ++ lib.optionals stdenv.hostPlatform.isDarwin [
```

also note that both the Darwins (aarch64/x86_64) will be using the same sdk version (11) soonish (the change is in staging but maybe a few weeks before it makes its way to master)